src/Http/Routing/src/Constraints/RegexRouteConstraint.cs

```diff
@@ -33,7 +34,7 @@ public RegexRouteConstraint(Regex regex)
     /// Constructor for a <see cref="RegexRouteConstraint"/> given a <paramref name="regexPattern"/>.
     /// </summary>
     /// <param name="regexPattern">A string containing the regex pattern.</param>
-    public RegexRouteConstraint(string regexPattern)
+    public RegexRouteConstraint([StringSyntax(StringSyntaxAttribute.Regex)] string regexPattern)
```

Comment:
```suggestion
    public RegexRouteConstraint([StringSyntax(StringSyntaxAttribute.Regex, RegexOptions.IgnoreCase | RegexOptions.CultureInvariant)] string regexPattern)
```

Reply:
What does this do?

Reply:
It informs tooling what options will be used with the pattern, as options can impact how the tooling handles the pattern.  For example, RegexOptions.IgnorePatternWhitespace impacts how the pattern is parsed, and RegexOptions.ECMAScript impacts how various constructs in the pattern are interpreted.  You could imagine tooling that tried to provide a nice English description of what the pattern does, and it would be impacted by IgnoreCase.  That said, at present I don't believe VS' syntax highlighting / error checking will care about IgnoreCase specifically.